### PR TITLE
Handle All-Designated HOV Lanes (e.g. hov:lanes:forward=designated|designated)

### PR DESCRIPTION
**Pull request by daniel-j-h**
# Issue

This implements handling HOV-only lanes similarly to `hov=designated`. For more details, check https://github.com/Project-OSRM/osrm-backend/issues/2929.

## Tasklist
 - [x] review
 - [x] adjust for for comments

cc @willwhite

### CODE REVIEW

**Comment by daniel-j-h** [comment]
@MoKob any chance we can still get this into `5.4` for @willwhite experiments?

File: features/car/access.feature [hunk 1]
```diff
@@ -156,6 +156,17 @@ Feature: Car - Restricted access
...
+    Scenario: Car - a way with all lanes HOV-designated is inaccessible by default (similar to hov=designated)
+        Then routability should be
+            | highway | hov:lanes:forward      | hov:lanes:backward     | forw | backw |
+            | primary | designated             | designated             |      |       |
+            | primary |                        | designated             | x    |       |
+            | primary | designated             |                        |      | x     |
+            | primary | designated\|designated | designated\|designated |      |       |
+            | primary | designated\|no         | designated\|no         | x    | x     |
+            | primary | yes\|no                | yes\|no                | x    | x     |
+            | primary |                        |                        | x    | x     |
```
Comment (MoKob) [inline features/car/access.feature]: for completion, shouldn't we add `hov:lanes` in combination with `oneway = yes`

File: profiles/car.lua [hunk 2]
```diff
@@ -238,6 +238,10 @@ function way_function (way, result)
     return
   end
 
+  -- default to driving mode, may get overwritten below
+  result.forward_mode = mode.driving
+  result.backward_mode = mode.driving
```
Comment (MoKob) [inline profiles/car.lua]: unrelated change?

File: profiles/car.lua [hunk 3]
```diff
@@ -250,6 +254,36 @@ function way_function (way, result)
     return
   end
 
+  -- also respect user-preference for HOV-only ways when all lanes are HOV-designated
+  local hov_lanes_forward = way:get_value_by_key("hov:lanes:forward")
```
Comment (MoKob) [inline profiles/car.lua]: how about `hov:lanes` and `oneway=yes`

File: profiles/car.lua (hunk 2 above)
Comment (daniel-j-h) [inline profiles/car.lua]: It's just moving setting the default travel mode to the top.

File: features/car/access.feature (hunk 1 above)
Comment (daniel-j-h) [inline features/car/access.feature]: Yes, good catch. Doing this!

File: profiles/car.lua [hunk 4]
```diff
@@ -238,17 +238,69 @@ function way_function (way, result)
...
+
+    if hov_all_designated_backward then
+      if reverse then
+        result.forward_mode = mode.inaccessible
+      else
+        result.backward_mode = mode.inaccessible
+      end
+    end
+
+  end -- hov handling
```
Comment (daniel-j-h) [inline profiles/car.lua]: This is getting really ugly here. Maybe moving to a separate file..

**Comment by daniel-j-h** [comment]
@MoKob ready for a second round of review. I'll squash commits into a single one afterwards.
